src/gui/attributetable/qgsdualview.cpp

```diff
@@ -140,7 +140,19 @@ void QgsDualView::init( QgsVectorLayer *layer, QgsMapCanvas *mapCanvas, const Qg
   // create an empty form to find out if it needs geometry or not
   const QgsAttributeForm emptyForm( mLayer, QgsFeature(), mEditorContext );
 
-  const bool needsGeometry = !( request.flags() & QgsFeatureRequest::NoGeometry )
+  bool conditionalFormattingRulesNeedsGeometry { false };
+  const auto cConditionalStyles { mLayer->conditionalStyles()->rowStyles() };
+  for ( const auto &style : std::as_const( cConditionalStyles ) )
+  {
```

Comment:
How do you feel about moving this loop to QgsConditionalLayerStyles::rulesNeedGeometry() ? That'd make it easier to add a test for.

Reply:
done